packages/schemas/src/types/connector.ts

```diff
@@ -12,6 +14,13 @@ export type ConnectorResponse = Pick<
   Omit<BaseConnector<ConnectorType>, 'configGuard' | 'metadata'> &
   ConnectorMetadata & { isDemo?: boolean };
 
+export const connectorFactoryResponseGuard = z
+  .object({
+    type: z.nativeEnum(ConnectorType), // Omit<BaseConnector<ConnectorType>, 'configGuard' | 'metadata'>
+    isDemo: z.boolean().optional(),
+  })
+  .merge(connectorMetadataGuard);
+
 export type ConnectorFactoryResponse = Omit<
```

Comment:
Should the ConnectorFactoryResponse type infer from the new connectorFactoryResponseGuard?

Reply:
Good call, fixed.